sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -465,9 +483,8 @@ std::pair<RT::PiKernel, std::mutex *> ProgramManager::getOrCreateKernel(
     return Result;
   };
 
-  const RT::PiDevice PiDevice = getRawSyclObjImpl(Device)->getHandleRef();
   auto BuildResult = getOrBuild<PiKernelT, invalid_object_error>(
-      Cache, std::make_pair(KernelName, PiDevice), AcquireF, GetF, BuildF);
+      Cache, KernelName, AcquireF, GetF, BuildF);
```

Comment:
it is not obvious to me why removing device from the cache key is safe - can a kernel built for one device be reused for another - ?

Reply:
> it is not obvious to me why removing device from the cache key is safe - can a kernel built for one device be reused for another - ?

The kernel cache is built per program. The program depends on build options and target device.
Device is not used to get kerpel for program. So device is redundant key.